src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -264,6 +287,23 @@ public void Write(byte[] data)
 
         #region Private Methods
 
+        internal IPAddress GetLocalNIAddress(string rni)
+        {
+            IPAddress addr = null;
+            if (IPAddress.TryParse(rni, out addr))
+                return addr;
+
+            foreach (NetworkInterface ni in NetworkInterface.GetAllNetworkInterfaces()) {
+                if (ni.Name == rni) { // UnicastAddresses[1] will give ipv4 address of certain adapter
+                    addr = ni.GetIPProperties().UnicastAddresses[0].Address; //This will give ipv6 address of certain adapter
```

Comment:
I worry that the IPv6 address may be non functional.  I think that it should at least be checked to make sure it is not [link local](https://docs.microsoft.com/en-us/dotnet/api/system.net.ipaddress.isipv6linklocal?view=net-6.0#system-net-ipaddress-isipv6linklocal)